xml/hwbp_registers.xml

```diff
@@ -777,18 +777,30 @@
     <register name="Instruction Count" short="icount" address="0x7a1">
         This register is accessible as \RcsrTdataOne when \FcsrTdataOneType is 3.
 
-        If \FcsrIcountCount is not 0, then every instruction completed or
+        When \FcsrIcountCount is greater than 1, every instruction completed or
         trap taken from a mode where the trigger is enabled decrements
         \FcsrIcountCount by 1.
-        When
-        \FcsrIcountCount is decremented from 1 to 0 then \FcsrIcountPending
-        becomes set. When \FcsrIcountPending is set, the trigger fires just
+
+        When \FcsrIcountCount is 1, and an instruction is completed or trap
+        taken from a mode where the trigger is enabled, then \FcsrIcountPending
+        becomes set. In addition \FcsrIcountCount may become 0 as well.
+
+        The only exception to the above is when the instruction executed is a
+        write to the icount trigger. In that case \FcsrIcountPending may or may
```

Comment:
I think that we were generally going with "might or might not" or simply "might not" in order to avoid the ambiguous "may not" (which is really not ambiguous in this particular context with "may or" but I think it's best to use consistent phrases).